src/utils/transfer-details.ts

```diff
@@ -1,13 +1,17 @@
 import type { Account } from "../account";
+import { throwError } from "./throw-error";
 
 export class TransferDetails {
   public sender: Account;
   public recipient: Account | string;
   public amount: number;
+  public memo: string;
 
-  constructor(sender: Account, recipient: Account | string, amount: number) {
+  constructor(sender: Account, recipient: Account | string, amount: number, memo: string) {
     this.sender = sender;
     this.recipient = recipient;
     this.amount = amount;
+    if (memo.length > 64) throwError("Invalid memo!");
```

Comment:
.matches(/^[a-zA-Z0-9_ ]*$/, 'Memo can only contain alphanumeric characters, spaces, and underscores')
        .max(64, 'Memo cannot exceed 64 characters'),
(memo: string): string => memo.trim();


What do you think some of these validations are maybe not needed?
clients been clients, we cant trust them to follow all requirements without enforcing it

Reply:
Well I can add that in if you'd like, but the server should respond with an error if those requirements aren't met. It's your call